crates/turbopack-ecmascript/src/references/require_context.rs

```diff
@@ -0,0 +1,267 @@
+use std::collections::VecDeque;
+
+use anyhow::Result;
+use indexmap::IndexMap;
+use swc_core::{
+    common::DUMMY_SP,
+    ecma::ast::{Expr, KeyValueProp, Lit, ObjectLit, Prop, PropName, PropOrSpread},
+    quote, quote_expr,
+};
+use turbo_tasks::{
+    primitives::{RegexVc, StringVc},
+    Value, ValueToString, ValueToStringVc,
+};
+use turbo_tasks_fs::{DirectoryContent, DirectoryEntry, FileSystemPathVc};
+use turbopack_core::{
+    chunk::{ChunkableAssetReference, ChunkableAssetReferenceVc},
+    issue::{IssueSeverityVc, OptionIssueSourceVc},
+    reference::{AssetReference, AssetReferenceVc},
+    resolve::{
+        origin::{ResolveOrigin, ResolveOriginVc},
+        parse::RequestVc,
+        ResolveResult, ResolveResultVc,
+    },
+};
+
+use crate::{
+    chunk::EcmascriptChunkingContextVc,
+    code_gen::{CodeGenerateable, CodeGeneration, CodeGenerationVc},
+    create_visitor,
+    references::{
+        pattern_mapping::{PatternMappingVc, ResolveType::Cjs},
+        AstPathVc,
+    },
+    resolve::{cjs_resolve, try_to_severity},
+    CodeGenerateableVc,
+};
+
+#[turbo_tasks::value(transparent)]
+pub(crate) struct FlatDirList(Vec<(String, FileSystemPathVc)>);
+
+#[turbo_tasks::function]
+pub(crate) async fn list_dir(
+    dir: FileSystemPathVc,
+    recursive: bool,
+    filter: RegexVc,
+) -> Result<FlatDirListVc> {
+    let root = &*dir.await?;
+    let filter = &*filter.await?;
+
+    let mut queue = VecDeque::with_capacity(1);
+    queue.push_back(dir);
```

Comment:
Nit:
```suggestion
    let mut queue: VecDeque<_> = [dir].collect();
```

Reply:
I don't like that at all, but `VecDeque::from([dir])` seems to work